tools/freebayes/freebayes.xml

```diff
@@ -1,529 +1,627 @@
 <?xml version="1.0"?>
 <tool id="freebayes" name="FreeBayes" version="0.4.1">
-  <requirements>
-    <requirement type="package" version="0_9_20_b040236">freebayes</requirement>
-    <requirement type="package" version="0.1.18">samtools</requirement>
-  </requirements>
-  <description> - bayesian genetic variant detector</description>
-  <command>
+    <description> - bayesian genetic variant detector</description>
+    <requirements>
+        <requirement type="package" version="0_9_20_b040236">freebayes</requirement>
+        <requirement type="package" version="0.1.18">samtools</requirement>
+        <requirement type="package" version="4.1.0">gnu_awk</requirement>
+        <requirement type="package" version="20131122">gnu_parallel</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+    <command>
+<![CDATA[
     ##set up input files
 
     #set $reference_fasta_filename = "localref.fa"
-    
+
     #if str( $reference_source.reference_source_selector ) == "history":
-        ln -s "${reference_source.ref_file}" "${reference_fasta_filename}" &amp;&amp;
-        samtools faidx "${reference_fasta_filename}" 2&gt;&amp;1 || echo "Error running samtools faidx for FreeBayes" &gt;&amp;2 &amp;&amp;
+        ln -s "${reference_source.ref_file}" "${reference_fasta_filename}" &&
+        samtools faidx "${reference_fasta_filename}" 2>&1 || echo "Error running samtools faidx for FreeBayes" >&2 &&
     #else:
         #set $reference_fasta_filename = str( $reference_source.ref_file.fields.path )
     #end if
-    
+
     #for $bam_count, $input_bam in enumerate( $reference_source.input_bams ):
-        ln -s "${input_bam.input_bam}" "localbam_${bam_count}.bam" &amp;&amp;
-        ln -s "${input_bam.input_bam.metadata.bam_index}" "localbam_${bam_count}.bam.bai" &amp;&amp;
+        ln -s "${input_bam}" "b_${bam_count}.bam" &&
+        ln -s "${input_bam.metadata.bam_index}" "b_${bam_count}.bam.bai" &&
```

Comment:
I'm a little bit worried here about the maximum character limit of a shell. We can probably hit this with 1000 of files ...